projects/angular/components/ui-grid/src/components/ui-grid-custom-paginator/ui-grid-custom-paginator.component.ts

```diff
@@ -0,0 +1,60 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    HostBinding,
+    Inject,
+    Injectable,
+    Optional,
+    ViewEncapsulation,
+} from '@angular/core';
+import { MatFormFieldAppearance } from '@angular/material/form-field';
+import {
+    MatPaginatorDefaultOptions,
+    MatPaginatorIntl,
+    MAT_PAGINATOR_DEFAULT_OPTIONS,
+    _MatPaginatorBase,
+} from '@angular/material/paginator';
+
+@Injectable({ providedIn: 'root' })
```

Comment:
This does not have to be provided in `root`, pls take a look at how `UiGridIntl` is set up and exported. 
We'll also need to be able to provide it in consumers as an alternate of `MatPaginatorIntl` with specific localization logic.
```suggestion
@Injectable()
```